addons/pinniped/post-deploy/pkg/configure/helper.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package configure
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/client-go/kubernetes"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/vmware-tanzu/tanzu-framework/addons/pinniped/post-deploy/pkg/constants"
+
+	"github.com/vmware-tanzu/tanzu-framework/addons/pinniped/post-deploy/pkg/configure/supervisor"
+)
+
+// createOrUpdatePinnipedInfo creates Pinniped information or updates existing data.
+func createOrUpdatePinnipedInfo(ctx context.Context, pinnipedInfo supervisor.PinnipedInfo, k8sClientSet kubernetes.Interface) error {
+	var err error
+	zap.S().Info("Creating the ConfigMap for Pinniped info")
+	data, err := json.Marshal(pinnipedInfo)
+	if err != nil {
+		err = fmt.Errorf("could not marshal Pinniped info into JSON: %w", err)
+		return err
+	}
+	dataMap := make(map[string]string)
+	err = json.Unmarshal(data, &dataMap)
```

Comment:
nit: this is usually more canonical in go:

```go
if err := json.Unmarshal(data, &dataMap); err != nil {
 ...
}
```

Reply:
thanks for pointing this out!  Honestly, it's super helpful for me to know the canonical or widely accepted ways of doing things in different languages, so feel free to pick all the nits!